test/src/unit-max_tile_size.cc

```diff
@@ -0,0 +1,1237 @@
+    array.close();
+  };
+
+  auto write_array_3 = [&]() -> void {
+    tiledb::Context ctx;
+
+    // Prepare some data for the array
+    std::vector<int> data = {201};
+
+    // Open the array for writing and create the query.
+    tiledb::Array array(ctx, array_name, TILEDB_WRITE);
+    tiledb::Subarray subarray(ctx, array);
+    subarray.add_range(0, 1, 1).add_range(1, 1, 1);
+    tiledb::Query query(ctx, array);
+    query.set_layout(TILEDB_ROW_MAJOR)
+        .set_data_buffer("a", data)
+        .set_subarray(subarray);
+
+    // Perform the write and close the array.
+    query.submit();
+    array.close();
+  };
+
+  auto write_array_4 = [&]() -> void {
+    tiledb::Context ctx;
+
+    // Prepare some data for the array
+    std::vector<int> data = {202};
+
+    // Open the array for writing and create the query.
+    tiledb::Array array(ctx, array_name, TILEDB_WRITE);
+    tiledb::Subarray subarray(ctx, array);
+    subarray.add_range(0, 3, 3).add_range(1, 4, 4);
+    tiledb::Query query(ctx, array);
+    query.set_layout(TILEDB_ROW_MAJOR)
+        .set_data_buffer("a", data)
+        .set_subarray(subarray);
+
+    // Perform the write and close the array.
+    query.submit();
+    array.close();
+  };
+
+  tiledb::Context ctx;
+
+  uint64_t capi_max = 0; // init to avoid msvc warning as error miss inside CHECK()
+  uint64_t cppapi_max;
+  // Create and write array
+  create_array();
+  CHECK((capi_max=get_fragments_extreme(array_name)) == 0);
+  cppapi_max=cpp_get_fragments_extreme(array_name);
+  CHECK(cppapi_max == capi_max);
+  write_array_1();
+  CHECK((capi_max=get_fragments_extreme(array_name)) == 4 * sizeof(int32_t));
+  cppapi_max = cpp_get_fragments_extreme(array_name);
+  CHECK(cppapi_max == capi_max);
+  write_array_2();
+  CHECK((capi_max=get_fragments_extreme(array_name)) == 4 * sizeof(int32_t));
+  cppapi_max = cpp_get_fragments_extreme(array_name);
+  CHECK(cppapi_max == capi_max);
+  write_array_3();
+  CHECK((capi_max=get_fragments_extreme(array_name)) == 4 * sizeof(int32_t));
+  cppapi_max = cpp_get_fragments_extreme(array_name);
+  CHECK(cppapi_max == capi_max);
+  write_array_4();
+  CHECK((capi_max=get_fragments_extreme(array_name)) == 4 * sizeof(int32_t));
+  cppapi_max=cpp_get_fragments_extreme(array_name);
+  CHECK(cppapi_max == capi_max);
+
+  // consolidate
+  tiledb::Array::consolidate(ctx, array_name);
+
+  CHECK((capi_max=get_fragments_extreme(array_name)) == 4 * sizeof(int32_t));
+  cppapi_max=cpp_get_fragments_extreme(array_name);
+  CHECK(cppapi_max == capi_max);
+
+  // consolidate
+  tiledb::Array::vacuum(ctx, array_name);
+
+  CHECK(get_fragments_extreme(array_name) == 4 * sizeof(int32_t));
+}
+
+TEST_CASE_METHOD(
+    CAPI_MaxTileSizeFx,
+    "C++ API: Max fragment size, sparse, variable string dimension, fix attr",
+    "[capi][max-tile-size][sparse][var-dimension]") {
+std::string array_name("sparse_string_dimension");
+tiledb::Context ctx;
+
+  auto  create_array = [&]()-> void {
```

Comment:
I believe each array is unique and view having the unique items specific to their locality of reference rather than requiring arbitrary renaming (_1, _2 etc) to avoid collision after moving outside local sphere of reference.

Similarly, having unique arrays, the writes are also individualized, and think are better located within the local sphere of usage.

Reply:
We have other tests using helpers to create any schema type... This really makes the test cases hard to read.